Client/Frontend/Login/LoginListViewController.swift

```diff
@@ -282,31 +311,26 @@ extension LoginListViewController {
 
     override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
         if editingStyle == .delete {
-            guard let loginItem = loginEntries[safe: indexPath.row] else { return }
+            guard let credential = credentialList[safe: indexPath.row] else { return }
             
-            showDeleteLoginWarning(with: loginItem)
+            showDeleteLoginWarning(with: credential)
         }
     }
     
     override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
         return indexPath.section == Section.savedLogins.rawValue
     }
     
-    private func showDeleteLoginWarning(with loginItem: Login) {
+    private func showDeleteLoginWarning(with credential: PasswordForm) {
         let alert = UIAlertController(
             title: Strings.deleteLoginAlertTitle,
             message: Strings.Login.loginEntryDeleteAlertMessage,
             preferredStyle: .alert)
         
         alert.addAction(UIAlertAction(title: Strings.deleteLoginButtonTitle, style: .destructive, handler: { [weak self] _ in
-            let success = self?.profile.logins.removeLoginByGUID(loginItem.guid)
-            
-            success?.upon { result in
-                if result.isSuccess {
-                    self?.fetchLoginInfo()
-                } else {
-                    log.error("Error while deleting a login entry")
-                }
+            DispatchQueue.main.async {
```

Comment:
Is this main async needed?

Reply:
Both Add and Remove are async in brave-core side underneath the interface. Calling them without dispatch main on iOS side is instant crash :(

Reply:
This is the closure of `UIAlertAction` though which should be called on main

Reply:
Unless you need a runloop hop so that it doesn't execute immediately?

Reply:
No you are totally right, this is not needed hereat all.
Removed at https://github.com/brave/brave-ios/pull/5013/commits/baa9b3860ae2d76d3847c6ffdda2e7b4bf498a5d